src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2667,7 +2667,7 @@ public override MemberInfo[] GetMembers(BindingFlags bindingAttr)
             return members;
         }
 
-        public override InterfaceMapping GetInterfaceMap(Type ifaceType)
+        public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] Type ifaceType)
```

Comment:
Why `NonPublicMethods`?

Reply:
As per Michal's example:
```C#
using System;

interface IFoo
{
    public void DoWork() => Console.WriteLine("UU");
    internal void OtherWork() => Console.WriteLine("OO");
}

class Base : IFoo
{
    static void Main()
    {
        foreach (var v in typeof(Base).GetInterfaceMap(typeof(IFoo)).TargetMethods)
        {
            Console.WriteLine(v);
        }
    }
}
```

This actually prints out both methods, the public `DoWork` as well as the internal `OtherWork`... so to keep this working the same after trimming - we need to preserve both public and non-public methods.

Reply:
Thanks for the explanation.